File: 01-Higher-Lower-Heads-Tails/01.rb
#!/usr/local/bin/ruby -w

def headtails
    coin = rand(0..1)
    if coin.zero?
        puts 'Heads!'
    elsif coin == 1
        puts 'Tails!'
    end
end

def higherlower
    target = 100
    major = target
    minor = 0
    puts 'Think of a number between 1 and 100.'
    loop do
        print "Is it #{target}? Y/H/L "
        answer = gets.chomp
        if answer.downcase == 'y'
            puts 'I did it!'
            break
        else
            if answer.downcase == 'l'
                major = target
            elsif answer.downcase == 'h'
                minor = target
            end
            targetarray = Array(minor..major)
            target = targetarray[targetarray.length / 2]
            if major == minor
                puts 'Make up your mind!'
                break
            end
            if major - minor == 1
                puts 'Make up your mind!'
                break
            end
        end
    end
end

loop do
    print 'Please select: 1 for coin flip, 2 for higher/lower. '
    choice = gets.chomp
    if [1, 2].include? choice.to_i
        if choice == '1'
            headtails
        else
            higherlower
        end
    end
end

File: 01-Higher-Lower-Heads-Tails/01.py
#!/usr/bin/env python3
import random


def headtails():
    coinresult = random.randint(0, 1)
    if coinresult == 0:
        print("Heads!")
    elif coinresult == 1:
        print("Tails!")


def higherlower():
    target = 100
    major = target
    minor = 1
    print("Think of a number between 1 and 100.")
    while True:
        answer = input("Is it %d? Y/H(igher)/L(ower) " % target)
        if answer.lower() == "y":
            print("I did it!")
            break
        else:
            if answer.lower() == "l":
                major = target
            elif answer.lower() == "h":
                if target == 100:
                    print("The number would be out of range!")
                else:
                    minor = target
            targetarray = range(minor, major)
            try:
                target = targetarray[int(len(targetarray)//2)]
            except IndexError:
                print("Make up your mind!")
                break
            if major == minor:
                print("Make up your mind!")
                break
            if (major - minor == 1) and (minor != 1):
                print("Make up your mind!")
                break


if __name__ == "__main__":
    while True:
        choice = input("Please select: 1 for coin flip, 2 for higher/lower. ")
        if choice == "1" or choice == "2":
            break
        else:
            print("No valid input.", end=" ")
    if choice == "1":
        headtails()
    elif choice == "2":
        higherlower()
